File: API/Data/Entities/Company.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using API.Models;

namespace API.Data.Entities;

[Table("companies")]
public class Company
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [Column("company_name")]
    [MaxLength(60)]
    public string CompanyName { get; set; } = string.Empty;

    [Column("company_type")]
    public CompanyType CompanyType { get; set; } = CompanyType.SoleProprietorship;

    [Column("bulstat")]
    [MaxLength(50)]
    public string? Bulstat { get; set; }

    [Column("vat_number")]
    [MaxLength(50)]
    public string? VatNumber { get; set; }

    [Column("is_vat_registered")]
    public bool? IsVatRegistered { get; set; } = false;

    [Column("address")]
    [MaxLength(80)]
    public string? Address { get; set; }

    [Column("city")]
    [MaxLength(30)]
    public string? City { get; set; }

    [Column("postal_code")]
    [MaxLength(15)]
    public string? PostalCode { get; set; }

    [Column("phone")]
    [MaxLength(20)]
    public string? Phone { get; set; }

    [Column("email")]
    [MaxLength(50)]
    public string? Email { get; set; }

    [Column("is_company_owner")]
    public bool IsCompanyOwner { get; set; }

    [Column("warranty_default_months")]
    public int? WarrantyDefaultMonths { get; set; } = 12;

    [Column("subscription_plan")]
    public SubscriptionPlan SubscriptionPlan { get; set; } = SubscriptionPlan.FreeTrial;

    [Column("subscription_status")]
    public SubscriptionStatus SubscriptionStatus { get; set; } = SubscriptionStatus.Trial;

    [Column("trial_start_date")]
    public DateTime? TrialStartDate { get; set; }

    [Column("trial_end_date")]
    public DateTime? TrialEndDate { get; set; }

    [Column("is_subscription_active")]
    public bool? IsSubscriptionActive { get; set; } = true;

    [Column("is_active")]
    public bool? IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public virtual ICollection<ApplicationUser> Users { get; set; } = new List<ApplicationUser>();
    public virtual ICollection<Montage> Montages { get; set; } = new List<Montage>();
    public virtual ICollection<InventoryItem> InventoryItems { get; set; } = new List<InventoryItem>();
}

File: frontend/src/types/company.ts

import type { CompanyType } from './common';

export interface Company {
  id: string;
  company_name: string;
  company_type: CompanyType;
  bulstat?: string | null;
  vat_number?: string | null;
  is_vat_registered: boolean;
  address?: string | null;
  city?: string | null;
  postal_code?: string | null;
  phone?: string | null;
  email?: string | null;
  subscription_plan?: string | null;
  subscription_status?: string | null;
  trial_end_date?: string | null;
  is_subscription_active?: boolean | null;
  last_activity_at?: string | null;
  is_active: boolean;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface CompanyUser {
  id: string;
  email: string;
  first_name: string;
  middle_name?: string | null;
  last_name: string;
  full_name: string;
  phone_number?: string | null;
  roles: string[];
  is_active: boolean;
}

export interface CreateCompanyRequest {
  company_name: string;
  company_type: CompanyType;
  bulstat?: string | null;
  vat_number?: string | null;
  is_vat_registered?: boolean;
  address?: string | null;
  city?: string | null;
  postal_code?: string | null;
  phone?: string | null;
  email?: string | null;
}

export interface UpdateCompanyRequest extends Partial<CreateCompanyRequest> {}

export interface CreateCompanyUserRequest {
  email: string;
  password: string;
  first_name: string;
  middle_name?: string | null;
  last_name: string;
  phone_number?: string | null;
  address?: string | null;
  role?: string;
}

export interface UpdateSubscriptionRequest {
  subscription_plan: string;
  subscription_status: string;
  is_subscription_active: boolean;
}

export type SubscriptionPlan = 'FreeTrial' | 'Premium' | 'Free';
export type SubscriptionStatus = 'Trial' | 'Active' | 'Expired' | 'Cancelled' | 'Suspended';

export const SUBSCRIPTION_PLANS: { value: Exclude<SubscriptionPlan, 'FreeTrial'>; label: string; description: string }[] = [
  { value: 'Free', label: 'Free', description: 'Up to 2 active employees' },
  { value: 'Premium', label: 'Premium', description: 'Unlimited employees' },
];

export const SUBSCRIPTION_STATUSES: { value: SubscriptionStatus; label: string }[] = [
  { value: 'Trial', label: 'Trial' },
  { value: 'Active', label: 'Active' },
  { value: 'Expired', label: 'Expired' },
  { value: 'Cancelled', label: 'Cancelled' },
  { value: 'Suspended', label: 'Suspended' },
];
